content/blog/package-react-components/index.md

```diff
@@ -0,0 +1,776 @@
+
+Most of this post is also applicable to any piece of Front End Javascript code that is related
+to CSS, images, etc, such as [Angular][3] Components / Directives, [Vue][15] Components, etc.
+
+## To Inline or not to Inline
+
+In the context of Front End assets, inlining means to bake those assets directly
+into the Javascript code (bundle) generated by our bundlers or compilers such as Webpack or [Babel][4].
+
+CSS can be inlined without any third party library but to enable all regular CSS features
+a third party **CSS-in-JS** library is recommended such as [Emotion][5] or [Styled-components][6], more on this later.
+Images can be inlined by Webpack's [url-loader][11] in the form of [Data URL][100] with base 64 encoding.
+SVG can be inlined by turning an `.svg` file into a [React][1] Component (An SVG file is just a regular HTML tree)
+or by using Webpack's [inline-svg-loader][101]
+
+The main advantage of inlining is that after you bundle all your code base you end up with
+a single Javascript file that contains all the dependencies it needs to run, even CSS and Images.
+
+This makes _very_ easy to package and publish to [NPM][102] and it makes very easy
+for **Component Consumers** to _use_ your module.
+
+Using **CSS-in-JS** libraries like [Emotion][5] or [Styled-components][6] makes up for a very
+nice **Run-Time** **Theming** API. Since everything is Javascript your Themes are regular Javascript
+object that can be composed or broken down into parts as any other regular JS object,
+and these can be altered at Run-Time, Themes can be composed, and they provide
+a much simpler strategy than Build-Time CSS generated Themes like [Bootstrap][9] or [Ant Design][8].
+[Material UI React][7] uses a Run-Time Theme via a custom **CSS-in-JS** solution.
+
+Inlining is particularly useful for Component libraries or even UI Frameworks because
+they tend to be mostly CSS heavy and with a couple of SVGs at most, both of which are efficiently
+inlined.
+
+Images are a bit more problematic because they are inlined in [base 64][103] making them 30% bigger.
+
+In my experience, most Component Libraries do not tend to use an over abundance
+of images and if they do they tend to be rather small.
+
+Most high resolution images are dynamic content that are usually dealt by
+Applications and not libraries, although libraries can render images, they do not
+ship with those images since they are _Content_ provided by APIs
+or static files in the App.
+
+The main drawback of inlining is that you turn multiple files that can
+be fetched in parallel by the browser to a single larger file that cannot,
+this will become more noticeable in larger code bases such as Big Component Libraries
+or Applications.
+
+If you have a big enough library of components you might able to break it
+into independent chunks that can be dynamically imported by Applications,
+allowing more parallelism.
+This is actually not a super far fetched solution since dynamic imports
```

Comment:
_dynamic imports_ was one of those concepts that I really didn't know and so I had to go and search it a little to understand how they make the situation better.

Reply:
Added a link and text explaining a bit the concept. Good observation